paymentsheet/src/main/java/com/stripe/android/paymentsheet/flowcontroller/PaymentSelectionUpdater.kt

```diff
@@ -1,33 +1,75 @@
 package com.stripe.android.paymentsheet.flowcontroller
 
+import com.stripe.android.paymentsheet.PaymentSheet
 import com.stripe.android.paymentsheet.model.PaymentSelection
 import com.stripe.android.paymentsheet.state.PaymentSheetState
+import com.stripe.android.ui.core.forms.resources.LpmRepository
 
 internal object PaymentSelectionUpdater {
 
+    private var currentConfig: PaymentSheet.Configuration? = null
+
     fun process(
         currentSelection: PaymentSelection?,
         newState: PaymentSheetState.Full,
+        supportedPaymentMethods: List<LpmRepository.SupportedPaymentMethod>,
     ): PaymentSelection? {
-        return currentSelection?.takeIf { it.canBeUsedIn(newState) } ?: newState.paymentSelection
+        if (currentConfig != null && currentConfig != newState.config) {
+            return null
+        }
```

Comment:
@tillh-stripe does this look right to you? If the paymentsheet config has changed, then we want to clear the payment selection, right?

Reply:
Yep, that’d be the catch-all. I wonder if we can limit it to certain properties of the config (maybe only `customer` and the upcoming `billingDetailsCollectionConfiguration`), but let’s figure that out in the next few weeks.